### ISSUE TITLE
2.4.0 release

### ISSUE THREAD

**Issue by adamralph** [FakeItEasy/FakeItEasy#880]
Can be labelled **ready** when all other issues on this milestone are closed.
- [ ] run code analysis in VS in _Release_ mode and address violations (send a regular PR which must be merged before continuing)
- [ ] if necessary, change `VERSION_SUFFIX` on the [CI Server](http://teamcity.codebetter.com/admin/editBuildParams.html?id=buildType:bt929)
    to appropriate "-beta123" or "" (for non-betas) value and initiate a build
- [ ] check build
- edit draft release in [GitHub UI](https://github.com/FakeItEasy/FakeItEasy/releases):
  - [ ] ensure completeness of release notes, including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
  - [ ] attach main nupkg and/or analyzer nupkg, whichever have content to release
  - [ ] publish the release
- [ ] push nupkg(s) to NuGet
- [ ] de-list pre-release or superseded buggy NuGet packages if present
- [ ] tweet, mentioning contributors and post link as comment here for easy retweeting ;-)
- [ ] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [ ] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [ ] run `rake next_version[new_version]` to
  - create a pull request that changes the version in CommonAssemblyInfo.cs to the expected version (of form _xx.yy.zz_)
  - create a new draft GitHub Release
  - create a new milestone for the next release
  - create a new issue (like this one) for the next release, adding it to the new milestone
- [ ] close this milestone

**Comment by adamralph** [comment]
Shall we close this? I think it's highly unlikely we'll ever release a 2.4.0.

**Comment by blairconrad** [comment]
There's always the "reopen" button!

**Comment by adamralph** [comment]
Thanks. I'll close the milestone too.

**Comment by blairconrad** [comment]
RIGHT! Sorry.
